contracts/KinFoundationVestingTrustee.sol

```diff
@@ -0,0 +1,230 @@
+        4.86777830487640090174734031e26,
+        3.89422264390112072139787225e26,
+        3.1153781151208965771182978e26,
+        2.49230249209671726169463824e26,
+        1.99384199367737380935571059e26,
+        1.59507359494189904748456847e26,
+        1.27605887595351923798765478e26,
+        1.02084710076281539039012382e26,
+        8.1667768061025231231209906e25,
+        6.5334214448820184984967924e25,
+        5.226737155905614798797434e25,
+        4.1813897247244918390379472e25,
+        3.3451117797795934712303577e25,
+        2.6760894238236747769842862e25,
+        2.1408715390589398215874289e25,
+        1.7126972312471518572699432e25,
+        1.3701577849977214858159545e25,
+        1.0961262279981771886527636e25,
+        8.769009823985417509222109e24,
+        7.015207859188334007377687e24,
+        5.61216628735066720590215e24,
+        4.48973302988053376472172e24,
+        3.591786423904427011777376e24,
+        2.873429139123541609421901e24,
+        1.1493716556494166437687604e25
+    ];
+
+    // The frequency by which the grant will vest.
+    uint256 public constant VESTING_FREQUENCY = 1 days;
+
+    // Used for calculating daily vested funds in calculateVestedTokens().
+    uint256 private constant DAYS_IN_YEAR = 365;
+
+    event NewGrant(address indexed _from, uint256 _value);
+    event TokensUnlocked(address indexed _to, uint256 _value);
+    event GrantRevoked(address indexed _holder, uint256 _refund);
+
+    /// @dev Reverts if Kin token or Kin Foundation addresses are set to zero.
+    modifier initialized() {
+        require(kin != address(0) && kinFoundation != address(0));
+
+        _;
+    }
+
+    /// @dev Reverts if the grant has not been given yet.
+    modifier granted() {
+        require(startTime != 0);
+
+        _;
+    }
+
+    /// @dev Reverts if the grant has been given already.
+    modifier notGranted() {
+        require(startTime == 0);
+
+        _;
+   }
+
+    /// @dev Reverts if called by anyone besides Kin Foundation wallet address.
+    modifier onlyKinFoundation() {
+        require(msg.sender == kinFoundation);
+
+        _;
+    }
+
+    /// @dev Constructor that initializes the address of the KinToken contract.
+    /// @param _kin KinToken The address of the previously deployed KinToken smart contract.
+    /// @param _kinFoundation address The address of Kin Foundation wallet.
+    function KinFoundationVestingTrustee(KinToken _kin, address _kinFoundation) {
+        require(_kin != address(0));
+        require(_kinFoundation != address(0));
+
+        kin = _kin;
+        kinFoundation = _kinFoundation;
+    }
+
+    /// @dev Revoke the grant by invalidating members and transfer all remaining funds back to owner.
+    function revoke() external onlyOwner initialized granted {
+        // Don't allow to revoke if there isn't any KIN left.
+        // This means all tokens have already been unlocked.
+        require(kin.balanceOf(this) > 0);
```

Comment:
is zero balance really an error condition? It's quite similar to unlockVestedTokens below, which just returns if there's nothing to unlock.